src/main/java/org/opentripplanner/middleware/otp/OtpDispatcherResponse.java

```diff
@@ -1,36 +1,85 @@
 package org.opentripplanner.middleware.otp;
 
 
+import org.apache.commons.lang3.SerializationUtils;
 import org.opentripplanner.middleware.otp.response.Response;
+import org.opentripplanner.middleware.utils.JsonUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.Serializable;
+import java.net.URI;
+import java.net.http.HttpResponse;
 
 /**
  * An OTP dispatcher response represents the status code and body return from a call to an OTP end point e.g. plan
  */
 
-public class OtpDispatcherResponse {
+public class OtpDispatcherResponse implements Serializable {
+    private static final Logger LOG = LoggerFactory.getLogger(OtpDispatcherResponse.class);
+
+    /** Empty constructor used for testing */
+    private OtpDispatcherResponse() {}
+
+    public OtpDispatcherResponse(HttpResponse<String> otpResponse) {
+        requestUri = otpResponse.uri();
+        responseBody = otpResponse.body();
+        statusCode = otpResponse.statusCode();
+        LOG.debug("Response from OTP server: {}", toString());
+    }
+
+    /**
+     * Constructor used only for testing.
+     */
+    public OtpDispatcherResponse(String otpResponse) {
+        requestUri = URI.create("http://test.com");
+        responseBody = otpResponse;
+        statusCode = 200;
+        LOG.debug("Response from OTP server: {}", toString());
+    }
 
     /**
      * Status code. Status code returned with response from an OTP server.
      */
     public int statusCode;
 
+    /** URI (for the OTP server) that the request was sent to */
+    public URI requestUri;
+
     /**
      * Response Body. Response Body returned with response from an OTP server.
      */
     public String responseBody = null;
 
     /**
      * Response. POJO version of response from an OTP server.
+     * Do not persist in case these classes change. This should always be re-instantiated from responseBody if needed.
      */
-    public Response response = null;
+    public Response getResponse() {
+        return JsonUtils.getPOJOFromJSON(responseBody, Response.class);
```

Comment:
This call has been resulting in errors (errors converting OTP's response into POJOs, looking at some log files).
Errors caused here prevent the corresponding HttpClient from being closed and can trigger a "Too many files open" exception.

Reply:
To be addressed in #70.